### DISCUSSION TITLE
strange issue on mobile browser

### DISCUSSION THREAD

**Discussion by chenxizhang** [marp-team/marp#284] [General]
Hi 

I have a simple page like below, I found the first page had a strange issue on mobile browser
```
---
marp: true
headingDivider: 3
theme: xizhang
paginate: true
title: Microsoft 365 高手是怎样炼成的
---

#  Microsoft 365 <br /> 高手是怎样炼成的
> 陈 希章 https://365pro.xizhang.com
![bg fit left](images/microsoft-365.png)

```

it will work fine on PC browser
![image](https://user-images.githubusercontent.com/1996954/159105611-45bdfadf-44e9-42d2-a034-a2cefa575ad2.png)

But it will not work fine on Mobile browser, looks like the image and text can't display correctly.
![image](https://user-images.githubusercontent.com/1996954/159105645-dc0f2407-32c3-4257-be35-1c1668d86d3e.png)

Can anyone help to test on your device?

The page link is  https://365pro.xizhang.com

**Comment by chenxizhang** [comment]
This issue is only found on page 1.

**Comment by yhatt** [comment]
It's [a known bug of Safari browser itself](https://bugs.webkit.org/show_bug.cgi?id=23113), but [our helper library for Marpit framework](https://github.com/marp-team/marpit-svg-polyfill) also does not apply polyfill correctly because your first slide seems to be changed the DOM structure by #272.

If you are using the plugin from #272, try to swap the order of the slide element and injected scripts.

```diff
 // If it is the first slide, prepend <script> tag before the rendered tag.
-if (meta && meta.marpitSlide === 0) return script + rendered
+if (meta && meta.marpitSlide === 0) return rendered + script
```

**Reply by chenxizhang** [reply]
WOW, you are a magician, I use `if (meta && meta.marpitSlide === 0) return rendered + script` and it works now.

**Comment by chenxizhang** [comment]
@yhatt sorry to bother you. The problem still there and even more complicate. when I open the page on mobile browser, it will refresh again and again, then crash.

Would you please help to take a look into this problem? please open https://365pro.xizhang.com/  or https://365pro.xizhang.com/excel.html

**Reply by yhatt** [reply]
@chenxizhang We cannot do anything for you because it is due to [a long-standing bug of WebKit for over 12 years](https://bugs.webkit.org/show_bug.cgi?id=23113). See more details in https://github.com/marp-team/marp-core/issues/195.

> In many cases, it brings crash in the big HTML slide deck as you shown. However, just rendering a few Marp slides like [marp.app](https://marp.app) seems to make crash in sometimes too.
> 
> Apple denies 3rd-party rendering engine in iOS app so all rendering engines are the same WebKit in any iOS browsers (Safari, Firefox, Chrome, Edge, Brave, etc...). For using Marp slides in iOS stably, you have to convert into the stable format like PDF, or raise your voice to Webkit bug tracker.
>
> _— https://github.com/marp-team/marp-core/issues/195#issuecomment-702288942_

An upcoming WebKit improvement described in https://blogs.igalia.com/nzimmermann/posts/2021-10-29-layer-based-svg-engine/ is still in progress, and I expect it requires more time to make stable rendering (about 2-3 years).

For now, please use PDF instead if you have to deliver large slide to users who use Safari/WebKit browser.

**Reply by chenxizhang** [reply]
ok, thanks for the clarify
